OpenRA.Game/Widgets/Widget.cs

```diff
@@ -182,6 +182,28 @@ public class ChromeLogic : IDisposable
 		protected virtual void Dispose(bool disposing) { }
 	}
 
+	public struct WidgetBounds
+	{
+		public int X, Y, Width, Height;
+		public readonly int Left => X;
+		public readonly int Right => X + Width;
+		public readonly int Top => Y;
+		public readonly int Bottom => Y + Height;
+
+		public WidgetBounds(int x, int y, int width, int height)
+		{
+			X = x;
+			Y = y;
+			Width = width;
+			Height = height;
+		}
+
+		public static explicit operator Rectangle(WidgetBounds src)
```

Comment:
Eww. I'd rather expose any helpers we needed, `IntersectsWith`, `Contains`, etc directly for ergonomic use and convert to a Rectangle internally if we're interested in reusing the logic.

Reply:
The issue is that `Game.Renderer.EnableScissor` requires a `Rectangle` and can't (IMO) cleanly be changed otherwise - and it seems a bit redundant to add those methods if we still need to convert to a `Rectangle` afterwards anyway.

I can replace this with to a `.ToRectangle()` method if that would be preferrable.

Reply:
Fair. Yes I'd take a `ToRectangle`.

Reply:
`Game.Renderer.EnableScissor` is only used by the widget code right? Why can't the signature be changed?

Reply:
its a core engine function that doesn't have visibility of the mod-defined `WidgetBounds` type.